src/dialectic/core.py

```
@@ -0,0 +1,305 @@
+            parts: list[str] = []
+
+            if not explicit_repr.is_empty():
+                parts.append(explicit_repr.format_as_markdown(include_ids=False))
+
+            if not derived_repr.is_empty():
+                # Include IDs for derived so agent can use get_reasoning_chain
+                parts.append(derived_repr.format_as_markdown(include_ids=True))
+
+            return "\n".join(parts)
+
+        except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)